Automattic/jetpack
# WAF: Handle share data option to write to log

#### Changes proposed in this Pull Request:

* With this patch we'll only write to the block log if the share data toggle is enabled.

#### Jetpack product discussion


p1651144025906559-slack-C029WFNV69M

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Enable the waf module.
* Try a request that the waf will block with the share data option turned off.
* Verify that nothing was written to the blocklog file.
* Turn the share data option on.
* Try the same request and verify that the blocklog file was created and the block was logged correctly.
* Bonus points: Verify that the bootstrap code is exporting the contents of the option correctly.
